### PR TITLE
syncer(dm): deprecate auto-fix-gtid

### LINKED ISSUE #3685: deprecate `auto-fix-gtid` of DM
### Is your feature request related to a problem?

seems this config has almost no effects because after retry `reSyncBinlog` will probably meet same "binlog purged" error. The only chance `reSyncBinlog` will succeed is after retry upstream switch happens, the new upstream does not lack GTID of its server_uuid and can continue replication. However, upstream switch will cause network IO error, which is handled differently.

Also, related functions like `Syncer.reSyncBinlog` could also be removed

### Describe the feature you'd like

as title. we can mark this field as deprecated and raise warning when set it.

### Describe alternatives you've considered

_No response_

### Teachability, Documentation, Adoption, Migration Strategy

_No response_

### PR DESCRIPTION
**Pull request by lance6716**
Signed-off-by: lance6716 <lance6716@gmail.com>

### What problem does this PR solve?

Issue Number: close #3685

### What is changed and how it works?

apart from #3685 , this PR also unify the function "RedirectStreamer" and "ResetReplicationSyncer"

### Check List 

Tests 

 - [WIP] add an unit test about `adjust`

### Release note 

```release-note
deprecate auto-fix-gtid in source config
```

### CODE REVIEW

**Review by lichunzhu — COMMENTED** [review]
Do we need to remove this?
https://github.com/pingcap/tiflow/blob/90ee38a6e040771db9bf09bbb116edb1deb1ef5e/dm/dm/worker/source_worker.go#L1063
Do we need to deprecate this?
https://github.com/pingcap/tiflow/blob/90ee38a6e040771db9bf09bbb116edb1deb1ef5e/dm/dm/config/task.go#L314

File: dm/syncer/streamer_controller.go [hunk 1]
```diff
@@ -241,15 +239,6 @@ func (c *StreamerController) resetReplicationSyncer(tctx *tcontext.Context, loca
 	return err
 }
 
-// RedirectStreamer redirects the streamer's begin position or gtid.
-func (c *StreamerController) RedirectStreamer(tctx *tcontext.Context, location binlog.Location) error {
-	c.Lock()
-	defer c.Unlock()
-
-	tctx.L().Info("redirect streamer", zap.Stringer("location", location))
```
Comment (lichunzhu) [inline dm/syncer/streamer_controller.go:249]: Do we need to add a log before L194?

**Review by lichunzhu — COMMENTED** [review]
Rest LGTM

**Review by lichunzhu — APPROVED** [review]
LGTM

**Comment by lance6716** [comment]
ping @GMHDBJD

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Comment by lance6716** [comment]
/hold

some CI failure is bad

**Pull request merged**